core/spec/models/spree/promotion_spec.rb

```diff
@@ -64,6 +65,10 @@
     it "should delete rules" do
       expect(Spree::PromotionRule.count).to eq(0)
     end
+
+    it "should delete codes" do
+      expect(Spree::PromotionCode.count).to eq(0)
```

Comment:
it could be nice to refactor this specs in an 'expect to change' fashion, but that's just a nice to have since the original approach isn't your code.